src/deploy.rs

```diff
@@ -168,148 +152,98 @@ impl DeployCmd {
             AnsibleInventoryType::Genesis,
-            Some(self.build_node_extra_vars_doc(None, None)?),
+            Some(self.build_node_extra_vars_doc(options, None, None)?),
         )?;
         print_duration(start.elapsed());
         Ok(())
     }
 
-    pub async fn provision_faucet(&self, genesis_multiaddr: &str) -> Result<()> {
-        let start = Instant::now();
-        println!("Running ansible against genesis node to deploy faucet...");
-        self.testnet_deploy.ansible_runner.run_playbook(
-            AnsiblePlaybook::Faucet,
-            AnsibleInventoryType::Genesis,
-            Some(self.build_faucet_extra_vars_doc(genesis_multiaddr)?),
-        )?;
-        print_duration(start.elapsed());
-        Ok(())
-    }
-
-    pub async fn provision_safenode_rpc_client(&self, genesis_multiaddr: &str) -> Result<()> {
-        let start = Instant::now();
-        println!("Running ansible against genesis node to start safenode_rpc_client service...");
-        self.testnet_deploy.ansible_runner.run_playbook(
-            AnsiblePlaybook::RpcClient,
-            AnsibleInventoryType::Genesis,
-            Some(self.build_safenode_rpc_client_extra_vars_doc(genesis_multiaddr)?),
-        )?;
-        print_duration(start.elapsed());
-        Ok(())
-    }
-
-    pub async fn provision_sn_auditor(&self, genesis_multiaddr: &str) -> Result<()> {
+    async fn provision_nodes(
+        &self,
+        options: &DeployOptions,
+        initial_contact_peer: &str,
+        node_type: NodeType,
+    ) -> Result<()> {
         let start = Instant::now();
-        println!("Running ansible against auditor machine to start sn_auditor service...");
-        self.testnet_deploy.ansible_runner.run_playbook(
-            AnsiblePlaybook::Auditor,
-            AnsibleInventoryType::Auditor,
-            Some(
-                self.build_sn_auditor_extra_vars_doc(
-                    genesis_multiaddr,
-                    self.beta_encryption_key
-                        .as_ref()
-                        .unwrap_or(&DEFAULT_BETA_ENCRYPTION_KEY.to_string()),
-                )?,
-            ),
-        )?;
-        print_duration(start.elapsed());
-        Ok(())
-    }
+        let inventory_type = match node_type {
+            NodeType::Bootstrap => AnsibleInventoryType::BootstrapNodes,
+            NodeType::Normal => AnsibleInventoryType::Nodes,
+        };
 
-    pub async fn provision_remaining_nodes(&self, initial_contact_peer: &str) -> Result<()> {
-        let start = Instant::now();
-        self.testnet_deploy.ansible_runner.run_playbook(
+        self.ansible_runner.run_playbook(
             AnsiblePlaybook::Nodes,
-            AnsibleInventoryType::Nodes,
+            inventory_type,
             Some(self.build_node_extra_vars_doc(
+                options,
                 Some(initial_contact_peer.to_string()),
-                Some(self.node_count),
+                Some(options.node_count),
```

Comment:
The bootstrap node_count should be wayy lower than the normal node count. Should we provide a option to specify the bootstrap node_count? I think @joshuef used 1 bootstrap node per machine.